.github/workflows/tests.yml

```
@@ -76,6 +76,7 @@ jobs:
     - uses: actions/checkout@85e6279cec87321a52edac9c87bce653a07cf6c2
       with:
         persist-credentials: false
+        ref: ${{ github.event.pull_request.head.sha || github.event.head_commit.id || github.event.pull_request.head.ref }}
```

```suggestion
        ref: ${{ github.event.pull_request.head.sha }}
```
Just this should be fine IMO, and more predictable.